### PR TITLE
added HoverLine package

### PR DESCRIPTION
**Pull request by nexional**
- [x] I'm the package's author and/or maintainer.
- [x] I have have read [the docs][1].
- [x] I have tagged a release with a [semver][2] version number.
- [x] My package repo has a description and a README describing what it's for and how to use it.
- [x] My package doesn't add context menu entries. *
- [x] My package doesn't add key bindings. **
- [x] Any commands are available via the command palette.
- [x] Preferences and keybindings (if any) are listed in the menu and the command palette, and open in split view.
- [ ] If my package is a syntax it doesn't also add a color scheme. ***
- [ ] I use [.gitattributes][3] to exclude files from the package: images, test files, sublime-project/workspace.

My package is ...
HoverLine: Sublime Text package to hover over a line to see line in its entirety

[1]: https://packagecontrol.io/docs/submitting_a_package
[2]: https://semver.org
[3]: https://www.git-scm.com/docs/gitattributes#_export_ignore

### CODE REVIEW

**Comment by kaste** [comment]
(Disclaimer: I'm not an official reviewer here.)

- `word_wrap` can have 3 states: true, false, and "auto".  Typically (by default) it should be "auto" which is "trueish" and therefore your plugin bails out.   I would bail out only for the literal `True` value and otherwise check `view.match_selector(0, "source") which is roughly what "auto" is about and gives practical good results.

E.g.

```
        if not self.is_enabled() or view.settings().get('word_wrap') is True:
            return

        if not view.match_selector(0, "source"):
            return
```

- the plugin is written in a *very* idiosyncratic style. This adds an unnecessary burden to review IMO and you should consider just using an auto-formatter like `black` even though such formatter produce completely boring source code.

- the usage of `locals()` is over the top here, you can just access the variables by their name. E.g. `locals()['toggle_package']`is just `toggle_package`.  E.g. `if True not in locals().values(): return` does nothing here.  

- a settings change observer is unnecessary here because the callback `is_enabled` has no side-effect -- it is just a getter.  

Other than that; I think this plugin makes some sense and it appears useful.

**Comment by kaste** [comment]
Lastly:  I would probably make the core functionality a `TextCommand` (which we can bind to a key or so) so that if we disbale `on_hover` we could still trigger it.  Mainly for keyboard only users.

File: repository/h.json [hunk 1]
```diff
@@ -1020,6 +1020,18 @@
 				}
 			]
 		},
+		{
+			"name": "HoverLine",
+			"details": "https://github.com/nexional/HoverLine",
+			"author": "VK",
```
Comment (braver) [inline repository/h.json:1026]: Who is VK?

**Comment by braver** [comment]
Please respond to the feedback so we can continue the review.

File: repository/h.json (hunk 1 above)
Comment (nexional) [reply repository/h.json:1026]: Will fix it to userid

**Comment by kaste** [comment]
Also remove repository/h.json.orig

**Pull request closed without merging**